stage.go

```diff
@@ -1,15 +1,40 @@
 package progress
 
+import "sync/atomic"
+
 type Stager interface {
 	Stage() string
 }
 
 type Stage struct {
-	Current string
+	currentStage atomic.Value
+}
+
+func (s *Stage) getCurrentStage() string {
+	return s.currentStage.Load().(string)
+}
+
+func (s *Stage) setCurrentStage(newStage string) {
+	s.currentStage.Store(newStage)
+}
+
+func (s *Stage) Stage() string {
+	return s.getCurrentStage()
+}
+
+func (s *Stage) Set(newStage string) {
+	s.setCurrentStage(newStage)
 }
 
-func (s Stage) Stage() string {
-	return s.Current
+func NewStage(current string) *Stage {
+	result := Stage{}
+	result.currentStage.Store(current)
+	return &result
+}
+
+type StagerSettable interface {
```

Comment:
there currently isn't a need to expose out setter interfaces since the interfaces are meant for consumers which often are read only.